### PR TITLE
ENH: verbose arg in set_eeg_reference

### PR DESCRIPTION
**Pull request by mmagnuski**
I often use `progressbar.ProgressBar` to track progress in long-running analysis script. However, `set_eeg_reference` is verbose (without verbose argument) - which breaks the progressbar into separate lines ( 😞 ). I think there should be a verbose argument in these functions.

TODO:
- [x] add a test? (*not needed*)
- [x] maybe something is missing? (*seems not*)
- [x] argh, and I forgot about adding verbose to the docstring... will do

### CODE REVIEW

File: mne/channels/channels.py [hunk 1]
```diff
@@ -237,7 +237,8 @@ def _check_set(ch, projs, ch_type):
 class SetChannelsMixin(object):
     """Mixin class for Raw, Evoked, Epochs."""
 
-    def set_eeg_reference(self, ref_channels=None):
+    @verbose
+    def set_eeg_reference(self, ref_channels=None, verbose=None):
```
Comment (Eric89GXL) [inline mne/channels/channels.py]: you need to update the arg list

File: mne/channels/channels.py [hunk 2]
```diff
@@ -279,7 +280,8 @@ def set_eeg_reference(self, ref_channels=None):
         mne.set_bipolar_reference
         """
         from ..io.reference import set_eeg_reference
-        return set_eeg_reference(self, ref_channels, copy=False)[0]
+        return set_eeg_reference(self, ref_channels, copy=False,
+                                 verbose=verbose)[0]
```
Comment (Eric89GXL) [inline mne/channels/channels.py]: you don't need to pass `verbose`  to the underlying function, so just undo this change

File: mne/io/reference.py [hunk 3]
```diff
@@ -231,7 +231,8 @@ def add_reference_channels(inst, ref_channels, copy=True):
     return inst
 
 
-def set_eeg_reference(inst, ref_channels=None, copy=True):
+@verbose
+def set_eeg_reference(inst, ref_channels=None, copy=True, verbose=None):
```
Comment (Eric89GXL) [inline mne/io/reference.py]: This is still useful to have as an option. But again need to update the arg list

File: mne/channels/channels.py (hunk 2 above)
Comment (mmagnuski) [inline mne/channels/channels.py]: oh, I didn't know, thanks!

File: mne/channels/channels.py (hunk 1 above)
Comment (Eric89GXL) [inline mne/channels/channels.py]: No, `verbose` is thoroughly tested elsewhere

Comment (mmagnuski) [inline mne/channels/channels.py]: 😌

**Comment by Eric89GXL** [comment]
Thanks @mmagnuski

**Comment by mmagnuski** [comment]
Thanks!
